lib/definition.js

```diff
@@ -0,0 +1,289 @@
+'use strict'
+
+const assert = require('assert')
+const has = require('./has.js')
+const MissingEnvError = require('./missing-env-error.js')
+const {
+  TRUE_VALUES,
+  FALSE_VALUES
+} = require('./boolean-values.js')
+
+const VALID_BOOLEANS = new Set([
+  ...Array.from(TRUE_VALUES)
+, ...Array.from(FALSE_VALUES)
+])
+
```

Comment:
Could be useful to have comments documenting RegExpError, EnumError, MissingEnvError, DocWriter etc. Could take from the README above

Reply:
Hmm, well, comments should be used sparingly to describe things that may be puzzling.  In these cases, the naming is pretty clear on what these things are for.  We should always try to make the code speak for itself.  Comments can become out-of-date and just as hard to manage as code (who wrote the comment?  Is it still valid? etc)

This does bring up a good point though, and that's that the documentation should contain the details from these things, and I don't see that in the `README`.  So thanks, I'll add it!

Reply:
I found some gaps in the custom errors.

* Added a `name` for `EnumError` which now matches the others
* Fixed the tests to check for the additional properties like `expected`, `actual` and `env`
* Added these errors to the `README.md`
* Indicated the errors that can be thrown by `validateEnvVars`